samples/attested_tls/client/enc/openssl_client.cpp

```diff
@@ -0,0 +1,308 @@
+// Copyright (c) Open Enclave SDK contributors.
+// Licensed under the MIT License.
+
+#include <errno.h>
+#include <openenclave/enclave.h>
+#include <openssl/bio.h>
+#include <openssl/err.h>
+#include <openssl/pem.h>
+#include <openssl/ssl.h>
+#include <openssl/x509.h>
+#include <openssl/x509_vfy.h>
+#include <string.h>
+#include "../../common/openssl_utility.h"
+
+#include <arpa/inet.h>
+#include <netdb.h>
+#include <netinet/in.h>
+#include <sys/socket.h>
+#include <unistd.h>
+
+int verify_callback(int preverify_ok, X509_STORE_CTX* ctx);
+
+extern "C"
+{
+    int launch_tls_client(char* server_name, char* server_port);
+};
+
+// This routine conducts a simple HTTP request/response communication with
+// server
+int communicate_with_server(SSL* ssl)
+{
+    unsigned char buf[200];
+    int ret = 1;
+    int error = 0;
+    int len = 0;
+    int bytes_written = 0;
+    int bytes_read = 0;
+
+    // Write an GET request to the server
+    printf(TLS_CLIENT "-----> Write to server:\n");
+    len = snprintf((char*)buf, sizeof(buf) - 1, CLIENT_PAYLOAD);
+    while ((bytes_written = SSL_write(ssl, buf, (size_t)len)) <= 0)
+    {
+        error = SSL_get_error(ssl, bytes_written);
+        if (error == SSL_ERROR_WANT_WRITE)
+            continue;
+        printf(TLS_CLIENT "Failed! SSL_write returned %d\n", error);
+        ret = bytes_written;
+        goto done;
+    }
+
+    printf(TLS_CLIENT "%d bytes written\n", bytes_written);
+
+    // Read the HTTP response from server
+    printf(TLS_CLIENT "<---- Read from server:\n");
+    do
+    {
+        len = sizeof(buf) - 1;
+        memset(buf, 0, sizeof(buf));
+        bytes_read = SSL_read(ssl, buf, (size_t)len);
+        if (bytes_read <= 0)
+        {
+            int error = SSL_get_error(ssl, bytes_read);
+            if (error == SSL_ERROR_WANT_READ)
+                continue;
+
+            printf(TLS_CLIENT "Failed! SSL_read returned error=%d\n", error);
+            ret = bytes_read;
+            break;
+        }
+
+        printf(TLS_CLIENT " %d bytes read\n", bytes_read);
+        // check to to see if received payload is expected
+        if ((bytes_read != SERVER_PAYLOAD_SIZE) ||
+            (memcmp(SERVER_PAYLOAD, buf, bytes_read) != 0))
+        {
+            printf(
+                TLS_CLIENT "ERROR: expected reading %lu bytes but only "
+                           "received %d bytes\n",
+                SERVER_PAYLOAD_SIZE,
+                bytes_read);
+            ret = bytes_read;
+            goto done;
+        }
+        else
+        {
+            printf(TLS_CLIENT
+                   " received all the expected data from server\n\n");
+            ret = 0;
+            break;
+        }
+        printf(TLS_CLIENT
```

Comment:
This line is unreachable. The `if` block will `goto done` whereas the `else` block will `break` out of the while loop.